src/main/java/fr/insee/lunatic/model/flat/ComponentType.java

```diff
@@ -83,35 +87,54 @@
         @JsonSubTypes.Type(value = Textarea.class, name = "Textarea"),
         @JsonSubTypes.Type(value = Suggester.class, name = "Suggester"),
         @JsonSubTypes.Type(value = Text.class, name = "Text"),
+        @JsonSubTypes.Type(value = FilterDescription.class, name = "FilterDescription"),
         @JsonSubTypes.Type(value = Accordion.class, name = "Accordion"),
 })
 @Getter
 @Setter
 public abstract class ComponentType {
 
+    /** Component identifier. */
+    @JsonProperty(required = true)
+    protected String id;
+
+    /** Property that defines the type of component in the serialized object. */
+    protected ComponentTypeEnum componentType;
+
+    /** Displayed label of the component. */
     protected LabelType label;
 
     /** Component description (concept introduced in Lunatic v3). */
     private LabelType description;
 
+    /** Declarations to display with the component. */
     @JsonInclude(Include.NON_EMPTY)
     protected List<DeclarationType> declarations;
+
+    /** Filter applied on the component. */
     protected ConditionFilterType conditionFilter;
+
+    /** Controls applied on the component. */
     @JsonInclude(Include.NON_EMPTY)
     protected List<ControlType> controls;
+
+    /** Object that indicates in which sequence/subsequence belongs the component.
+     * @deprecated Unused by Lunatic anymore, to be removed. */
+    @Deprecated(since = "5.1.0", forRemoval = true)
     protected Hierarchy hierarchy;
+
     protected ResponseType missingResponse;
 
     /** Collected or external variable names required to evaluate expressions used in the component properties.
      * @deprecated Binding dependencies at component level are not used anymore in Lunatic. */
     @JsonInclude(Include.NON_EMPTY)
-    @Deprecated(since = "3.4.0")
+    @Deprecated(since = "3.4.0", forRemoval = true)
```

Comment:
Please proceed carefullyt, this parameter is still being used

Reply:
`bindingsDependencies` still exists in CALCULATED variables

Reply:
@alicela confirms, we can depreciate this props